### DISCUSSION TITLE
How to get key of selected row without event?

### DISCUSSION THREAD

**Discussion by pawamoy** [Textualize/textual#3623] [Q&A] [answered]
I'm subclassing `Container` to create a `Column` class which is composed of a `DataTable`.
My `Column` class declares a binding on the `enter` key (with priority). Contrary to events, the action for this binding does not receive anything, and I don't see any attribute or method on the `DataTable` class that allows to get a row key given the datatable state only.

Well, I think I could do the following, but it seems extremely wasteful:

```python
row_data = dt.get_row_at(dt.cursor_row)
for key in dt.rows:
    if dt.get_row(key) == row_data:
        row_key = key
        break
```

Isn't there a way to get the row key of the currently selected row without iterating and comparing every row? :thinking: 

My code:

```python
class Column(Container):
    BINDINGS = [
        Binding("enter", "default", "Execute default action", priority=True),
    ]

    def compose(self):
        yield DataTable()

    # on_mount, update, etc.

    def action_default(self) -> None:
        datatable = self.query_one(DataTable)
        row_index = datatable.cursor_row
        row = datatable.get_row_at(row_index)
        # how to get row_key?
```

As for *why* I want the row key, it's because it is necessary if I want to remove the row with `dt.remove_row()`, for example.

**Comment by joshbduncan** [comment, marked as answer]
You can get the row and column keys using the line below...

```python
row_key, col_key = dt.coordinate_to_cell_key(dt.cursor_coordinate)
```

I remove the selected row in a project of mine using the function call below. Please note, I'm using `self` as a have a custom subclassed DataTable. 

```python
def action_remove_row(self) -> None:
    """Delete the currently selected row."""
    if not self.row_count:
        return
    row_key, _ = self.coordinate_to_cell_key(self.cursor_coordinate)
    self.remove_row(row_key)
```

**Reply by pawamoy** [reply]
Thanks! It wasn't obvious that the `CellKey` is actually a `(RowKey, ColumnKey)` tuple. The docs should definitely mention it, I'll see if I can open a PR :slightly_smiling_face:
